CRM/Utils/Token.php

```diff
@@ -1255,9 +1255,13 @@ public static function getTokenDetails(
           $pcm = CRM_Core_PseudoConstant::get('CRM_Contact_DAO_Contact', 'preferred_communication_method');
 
           // communication Preference
-          $contactPcm = explode(CRM_Core_DAO::VALUE_SEPARATOR,
-            $contactDetails[$contactID]['preferred_communication_method']
-          );
+          $contactPcm = $contactDetails[$contactID]['preferred_communication_method'];
+          // this part is deprecated as due to recent improvement, we get a array of
+          //  preferred communication values instead of string via Contact.Get API
+          if (!isset($contactPcm)) {
```

Comment:
I guess so - I'm inclined to try to get this replicated in a test before merging. Now I know the cause of it I can see it is a regression (caused by me as it turns out) but not one that is within-release or high-impact so I think it makes sense to slow it down & figure out a test